src/serializer/serializer.js

```diff
@@ -1245,6 +1323,7 @@ export class Serializer {
         for (let instance of instances) {
           let { functionValue, serializedBindings } = instance;
           let id = this._getValIdForReference(functionValue);
+          invariant(id.type === "Identifier");
```

Comment:
This is a bit surprising since _getValIdForReference specifies BabelNodeIdentifier as its return type.